File: flutter/test/log4agent_redactor_test.dart
import 'package:log4agent/log4agent.dart';
import 'package:test/test.dart';

void main() {
  test('redacts sensitive attributes', () {
    final result = Log4AgentRedactor.redactAttributes({
      'Authorization': 'Bearer abc',
      'symbol': 'BTC-USDT',
    });

    expect(result['Authorization'], Log4AgentRedactor.redacted);
    expect(result['symbol'], 'BTC-USDT');
  });

  test('redacts sensitive query params', () {
    final result = Log4AgentRedactor.redactUrl(
      'https://example.com/path?token=abc&symbol=BTC-USDT&code=123',
    );

    expect(
      result,
      'https://example.com/path?token=[REDACTED]&symbol=BTC-USDT&code=[REDACTED]',
    );
  });
}

File: kmp/log4agent/src/commonTest/kotlin/dev/log4agent/Log4AgentRedactorTest.kt
package dev.log4agent

import kotlin.test.Test
import kotlin.test.assertEquals
import kotlin.test.assertFalse

class Log4AgentRedactorTest {
    @Test
    fun redactsSensitiveAttributeKeys() {
        val redacted = Log4AgentRedactor.redactAttributes(
            mapOf(
                "Authorization" to "Bearer abc",
                "symbol" to "BTC-USDT",
            ),
        )

        assertEquals(Log4AgentRedactor.REDACTED, redacted["Authorization"])
        assertEquals("BTC-USDT", redacted["symbol"])
    }

    @Test
    fun redactsSensitiveQueryParams() {
        val redacted = Log4AgentRedactor.redactUrl(
            "https://example.com/login?token=abc&symbol=BTC-USDT&code=123",
        )

        assertFalse(redacted.contains("abc"))
        assertFalse(redacted.contains("123"))
        assertEquals(
            "https://example.com/login?token=[REDACTED]&symbol=BTC-USDT&code=[REDACTED]",
            redacted,
        )
    }
}

